src/RA_Defs.h

```diff
@@ -225,30 +255,382 @@ typedef DWORD			ARGB;
 #undef ASSERT
 #define ASSERT( x ) {}
 #endif
-	
+
 #ifndef UNUSED
 #define UNUSED( x ) ( x );
 #endif
 
-extern std::string Narrow(const wchar_t* wstr);
-extern std::string Narrow(const std::wstring& wstr);
-extern std::wstring Widen(const char* str);
-extern std::wstring Widen(const std::string& str);
 
-//	No-ops to help convert:
-//	No-ops to help convert:
-extern std::wstring Widen(const wchar_t* wstr);
-extern std::wstring Widen(const std::wstring& wstr);
-extern std::string Narrow(const char* str);
-extern std::string Narrow(const std::string& wstr);
 
-typedef std::basic_string<TCHAR> tstring;
 
 
+
+
+
+namespace ra {
+using cstring       = const char*;
+using cwstring      = const wchar_t*;
+using tstring       = std::basic_string<TCHAR>;
+
+template<typename CharT>
+using string_t = std::basic_string<CharT>;
+
+using DataStream     = std::basic_string<BYTE>;
+using ustring        = DataStream;
+using uofstream      = std::basic_ofstream<BYTE>;
+using uifstream      = std::basic_ifstream<BYTE>;
+using uostringstream = std::basic_ostringstream<BYTE>;
+using uistringstream = std::basic_istringstream<BYTE>;
+using uostream       = std::basic_ostream<BYTE>;
+using uistream       = std::basic_istream<BYTE>;
+
+using ByteAddress   = std::size_t;
+using DataPos       = std::size_t;
+using AchievementID = std::size_t;
+using LeaderboardID = std::size_t;
+using GameID        = std::size_t;
```

Comment:
i don't think any of these will ever be more than 32-bits, but since we're targeting 32-bit, size_t is 32-bits.

Reply:
Well size_t has been used throughout the project before I even touched it, it felt appropriate.

Reply:
https://github.com/RetroAchievements/RAIntegration/blob/master/src/RA_Defs.h#L203-L207

Looks like ByteAddress was already 64-bit and this actually shortens it to 32-bit.

Reply:
<code>unsigned long</code> isn't 64-bit, <code>unsigned long long</code> is. At least in VC++.